frontend/src/libs/components/header/header.tsx

```diff
@@ -6,10 +6,11 @@ import { type UserAuthResponse } from '~/packages/users/users.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  user: UserAuthResponse;
+  user: UserAuthResponse | null;
```

Comment:
Because header is shown only when user is logged in
```suggestion
  user: UserAuthResponse;
```

Reply:
![image](https://user-images.githubusercontent.com/101496324/227719714-4504e34d-78ba-4e72-a494-033fba4703a4.png)
method getCurrentUser return `user: UserAuthResponse | null` im my understanding method must return user, checking must be on startup, not in action. so based on this return type here `UserAuthResponse | null` too, or i dont understand something?

Reply:
My projects is a protected route -> protected route requires a logged in user -> in this case you're sure that user is not null

Reply:
I understand it, but in store user with type `UserAuthResponse | null`, so if I leave in Header component only `UserAuthResponse` there is an error, how can i resolve it? i'm confused

Reply:
just cast to `UserAuthResponse` in place where you need it